src/config/C4Reloc.cpp

```diff
@@ -69,14 +71,83 @@ bool C4Reloc::AddPath(const char* path, PathType pathType)
 	return true;
 }
 
+
+C4Reloc::const_iterator& C4Reloc::const_iterator::operator++()
+{
+	if (subdirIters.empty())
+	{
+		if ((*pathListIter).pathType == C4Reloc::PathType::PATH_IncludingSubdirectories)
+		{
+			DirectoryIterator subdir((*pathListIter).strBuf.getData());
+			if (*subdir)
+				subdirIters.emplace(subdir);
+			else
+				++pathListIter;
+		}
+		else
+		{
+			++pathListIter;
+		}
+	}
+	else // Currently in a subdir?
+	{
+		DirectoryIterator &currentSubdir = subdirIters.top();
+
+		if ((!*currentSubdir) || !*(++currentSubdir))
+		{
+			subdirIters.pop();
+			if (subdirIters.empty())
+				++pathListIter;
+		}
+		else
+		{
+			// Go deeper?
+			if (DirectoryExists(*currentSubdir))
+			{
+				DirectoryIterator subdir(*currentSubdir);
+				if (*subdir) // Make sure there is at least one file/subdir.
+					subdirIters.emplace(subdir);
+			}
+		}
+	}
+	return *this;
+}
+const C4Reloc::PathInfo & C4Reloc::const_iterator::operator*() const
+{
+	if (!subdirIters.empty())
+	{
+		const DirectoryIterator &currentSubdir = subdirIters.top();
+		temporaryPathInfo.reset(new C4Reloc::PathInfo(StdStrBuf(*currentSubdir), PathType::PATH_Regular));
+		return *temporaryPathInfo;
+	}
+	return *pathListIter;
+}
+
+bool operator==(const C4Reloc::const_iterator& a, const C4Reloc::const_iterator& b)
+{
+	if (!a.subdirIters.empty()) return false;
+	if (!b.subdirIters.empty()) return false;
+	if (a.pathListIter != b.pathListIter)
```

Comment:
You could reduce these three lines to `return a.pathListIter == b.pathListIter;`